homeassistant/components/esphome/ffmpeg_proxy.py

```diff
@@ -0,0 +1,178 @@
+"""HTTP view that converts audio from a URL to a preferred format."""
+
+import asyncio
+from collections import defaultdict
+from dataclasses import dataclass, field
+from http import HTTPStatus
+import logging
+from typing import Any
+
+from aiohttp import web
+from aiohttp.abc import AbstractStreamWriter, BaseRequest
+import voluptuous as vol
+
+from homeassistant.components.ffmpeg import FFmpegManager
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+
+from .const import DATA_FFMPEG_PROXY
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def async_allow_proxy_url(hass: HomeAssistant, device_id: str, url: str) -> None:
+    """Mark a URL as allowed for conversion by a specific device.
+
+    The URL will be removed from the allow list once it is requested in the HTTP
+    view.
+    """
+    data: FFmpegProxyData = hass.data[DATA_FFMPEG_PROXY]
+    data.device_urls[device_id].add(url)
```

Comment:
I would suggest that we generate a new unique URL instead of passing the URL around. (ie based on `ulid()`).